merlin/models/tf/prediction_tasks/classification.py

```diff
@@ -226,3 +226,6 @@ def from_schema(
 
     def call(self, inputs, training=False, **kwargs):
         return inputs
+
+    def compute_output_shape(self, input_shape):
+        self.pre.compute_output_shape(input_shape)
```

Comment:
This was related to the unit test `test_two_tower_advanced_options`. Adding a call method the logQ block raises an issue of `missing compute_output_shape` in the classification task when training a retrieval model (two-tower or MF) with an ItemRetrievalTask and LogQ correction. 

Thank you for the suggestion, I moved the implementation of the compute output shape to the `PredictionTask` and made sure we check if `task_block` or `pre` are specified to call their `compute_output_shape` methods, otherwise we return the original `input_shape`.